The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20932095808/job/60145395976) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/60145395976) [(plain)](https://github.com/rust-lang/rust/commit/769d7d6ce730666db5a6a1ea393b2efea493f9a5/checks/60145395976/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
............................................       (144/144)

======== tests/rustdoc-gui/notable-trait.goml ========

[ERROR] line 255: expected 0 elements, found 1: for command `assert-count: ("//*[@class='tooltip popover']", 0)`
    at <file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/test_docs/struct.NotableStructWithLongName.html#method.create_an_iterator_from_read>

======== tests/rustdoc-gui/search-result-display.goml ========

[WARNING] line 39: Delta is 0 for "x", maybe try to use `compare-elements-position` instead?

```

</details>

--- a/library/std/src/sys/thread/unix.rs
+++ b/library/std/src/sys/thread/unix.rs
@@ -45,6 +45,15 @@ impl Thread {
     // unsafe: see thread::Builder::spawn_unchecked for safety requirements
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     pub unsafe fn new(stack: usize, init: Box<ThreadInit>) -> io::Result<Thread> {
+        // FIXME: remove this block once wasi-sdk is updated with the fix from
+        // https://github.com/WebAssembly/wasi-libc/pull/716
+        // WASI does not support threading via pthreads. While wasi-libc provides
+        // pthread stubs, pthread_create returns EAGAIN, which causes confusing
+        // errors. We return UNSUPPORTED_PLATFORM directly instead.
+        if cfg!(target_os = "wasi") {
+            return Err(io::Error::UNSUPPORTED_PLATFORM);
+        }
+
         let data = init;
         let mut attr: mem::MaybeUninit<libc::pthread_attr_t> = mem::MaybeUninit::uninit();
         assert_eq!(libc::pthread_attr_init(attr.as_mut_ptr()), 0);

